packages/ra-core/src/dataProvider/useQueryWithStore.ts

```diff
@@ -120,7 +121,8 @@ const useQueryWithStore = (
     loaded: boolean;
 } => {
     const { type, resource, payload } = query;
-    const requestSignature = JSON.stringify({ query, options });
+    const version = useVersion(); // used to allow force reload
```

Comment:
Now even if the caller forgets to pass the version in the query options, the side effect will re-execute when a REFRESH_VIEW action is dispatched.